TiDB Operator panics when deploying a heterogeneous TiKV cluster

## Bug Report

**What version of Kubernetes are you using?**
<!-- You can run `kubectl version` -->

**What version of TiDB Operator are you using?**
<!-- You can run `kubectl exec -n tidb-admin {tidb-controller-manager-pod-name} -- tidb-controller-manager -V` -->

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
<!-- You can run `kubectl get sc` and `kubectl get pvc -n {tidb-cluster-namespace}` -->

**What's the status of the TiDB cluster pods?**
<!-- You can run `kubectl get po -n {tidb-cluster-namespace} -o wide` -->

**What did you do?**
<!-- If possible, provide a recipe for reproducing the error. How you installed tidb-operator and tidb-cluster. -->

- Install a normal TiDB cluster with PD, TiDB, and TiKV
- Install a heterogeneous TiKV cluster:

```
apiVersion: pingcap.com/v1alpha1
kind: TidbCluster
metadata:
  name: ns3
  namespace: ns2
spec:
  configUpdateStrategy: RollingUpdate
  discovery: {}
  enablePVReclaim: false
  imagePullPolicy: IfNotPresent
  cluster:
    name: ns2
  tikv:
    affinity: {}
    baseImage: pingcap/tikv
    config: |
      log-level = "info"
    evictLeaderTimeout: 1m
    imagePullPolicy: IfNotPresent
    maxFailoverCount: 3
    recoverFailover: true
    replicas: 1
    requests:
      storage: 10Gi
    storageClassName: local-storage
  tlsCluster: {}
  version: v5.0.0
```
**What did you expect to see?**

**What did you see instead?**
TiDB Operator panics with the follwoing error:
```
E0507 02:12:10.021688       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 475 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1b68300, 0x313d640)
        k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1b68300, 0x313d640)
        runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1.(*TidbCluster).TiDBStsDesiredOrdinals(0xc00047cf00, 0xc0000de000, 0x2252420)
        github.com/pingcap/tidb-operator@/pkg/apis/pingcap/v1alpha1/tidbcluster.go:491 +0x2d
github.com/pingcap/tidb-operator/pkg/manager/member.(*TidbClusterStatusManager).syncTiDBInfoKey(0xc00000e208, 0xc00047cf00, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/manager/member/tidbcluster_status_manager.go:206 +0x20a
github.com/pingcap/tidb-operator/pkg/manager/member.(*TidbClusterStatusManager).Sync(0xc00000e208, 0xc00047cf00, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/manager/member/tidbcluster_status_manager.go:56 +0x70
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).updateTidbCluster(0xc00087f100, 0xc00047cf00, 0xc0011f0c01, 0x143c09d)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:255 +0x789
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).UpdateTidbCluster(0xc00087f100, 0xc00047cf00, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:107 +0xc8
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).syncTidbCluster(...)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:159
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).sync(0xc000836c60, 0xc0005dc2a9, 0x7, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:155 +0x1f0
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).processNextWorkItem(0xc000836c60, 0x0)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:122 +0x100
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).worker(0xc000836c60)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:110 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000f4a060)
        k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000f4a060, 0x3b9aca00, 0x0, 0x1, 0xc000918cc0)
        k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000f4a060, 0x3b9aca00, 0xc000918cc0)
        k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:88 +0x4d
created by github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).Run
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:102 +0x1d9
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x158 pc=0x1405b6d]

goroutine 475 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1b68300, 0x313d640)
        runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1.(*TidbCluster).TiDBStsDesiredOrdinals(0xc00047cf00, 0xc0000de000, 0x2252420)
        github.com/pingcap/tidb-operator@/pkg/apis/pingcap/v1alpha1/tidbcluster.go:491 +0x2d
github.com/pingcap/tidb-operator/pkg/manager/member.(*TidbClusterStatusManager).syncTiDBInfoKey(0xc00000e208, 0xc00047cf00, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/manager/member/tidbcluster_status_manager.go:206 +0x20a
github.com/pingcap/tidb-operator/pkg/manager/member.(*TidbClusterStatusManager).Sync(0xc00000e208, 0xc00047cf00, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/manager/member/tidbcluster_status_manager.go:56 +0x70
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).updateTidbCluster(0xc00087f100, 0xc00047cf00, 0xc0011f0c01, 0x143c09d)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:255 +0x789
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).UpdateTidbCluster(0xc00087f100, 0xc00047cf00, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:107 +0xc8
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).syncTidbCluster(...)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:159
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).sync(0xc000836c60, 0xc0005dc2a9, 0x7, 0x0, 0x0)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:155 +0x1f0
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).processNextWorkItem(0xc000836c60, 0x0)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:122 +0x100
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).worker(0xc000836c60)
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:110 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000f4a060)
        k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000f4a060, 0x3b9aca00, 0x0, 0x1, 0xc000918cc0)
        k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000f4a060, 0x3b9aca00, 0xc000918cc0)
        k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:88 +0x4d
created by github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).Run
        github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:102 +0x1d9
```

--- a/pkg/manager/member/tidbcluster_status_manager.go
+++ b/pkg/manager/member/tidbcluster_status_manager.go
@@ -96,6 +96,10 @@ func getTidbAddr(key string) (addr string) {
 }
 
 func (m *TidbClusterStatusManager) syncTiDBInfoKey(tc *v1alpha1.TidbCluster) error {
+	if tc.Spec.TiDB == nil {
+		return nil
+	}
+
 	var pdEtcdClient pdapi.PDEtcdClient
 	var err error
 

